cEP-0029.md

```diff
@@ -0,0 +1,342 @@
+created that will allow users to quickly bootstrap their projects.
+
+
+## TOML syntax for coala configuration
+
+
+### Basic TOML concepts
+- The basic building blocks of toml are `key-value` pairs.
+- Values can be integers, booleans, arrays, floats,
+  date and time
+- A collection of key-value pairs form a `table`.
+  A table can be used to group rules into a section.
+- Array of tables can be used to list `collections`.
+eg:- collection of friends with their attributes
+
+### coala configuration concepts
+From the perspective of coala, only `tables` are of importance.
+Configuration concepts currently in use in coala:
+- Section as a collection of settings (key-value pairs)
+- Concept of inheritance in sections
+- Aspects and tastes
+
+Keeping the TOML concepts and coala configuration concepts
+in mind, TOML syntax for coala has been decided.
+
+- Settings shall be declared as key-value pairs
+- Sections shall be declared as tables
+- Inheritance shall be supported through the use of
+a key called `inherits`. Appending of an inherited setting
+shall be through a key called `append`.
+- Aspects from a section shall be declared in the form of
+  `[section_name.aspect_name]`
+
+
+### Aspects or settings as a group of settings
+The syntax [all.python] cannot be used to depict inheritance
+in toml as in coafile. This is because TOML parsers consider
+`python` to be a setting in the section `all`.
+This type of syntax is more suitable when a setting itself
+is defined by a group of settings within a section.
+```toml
+[all]
+a.b = 10
+a.c = 20
+a.d = 30
+b = 45
+```
+can also be represented as
+
+```toml
+[all]
+b = 10
+
+[all.a]
+b = 10
+c = 20
+d = 30
+```
+
+Since aspects are settings within a section and since aspects
+have their own group of settings, this syntax perfectly suits
+aspects.
+
+Aspects in coafile
+```text
+[all]
+files = **
+aspects = aspectname1, AspectName2 # case-insensitive
+# defining an aspect's taste
+aspectname1:aspect_taste = 80
+# we can define subaspect taste through its parent
+aspectname1:subaspect_taste = word1, word2, word3
+```
+
+is equivalent to
+
+```toml
+[all]
+files = '**'
+aspects = ['aspectname1', 'AspectName2'] # case-insensitive
+
+[all.aspectname1]
+aspect_taste = 80
+subaspect_taste = ['word1', 'word2', 'word3']
+```
+
+or
+
+```toml
+[all]
+files = '**'
+aspects = ['aspectname1', 'AspectName2'] # case-insensitive
+aspectname1.aspect_taste = 80
+AspectName2.subaspect_taste = ['word1', 'word2', 'word3']
+```
+in TOML.
+
+### Inheritance
+
+Inheritance in coafile
+```text
+[all]
+enabled = True
+overridable = 2
+ignore = vendor1/
+
+[all.section1]
+overridable = 3
+ignore += vendor2/
+other = some_value
+```
+
+is equivalent to
+
+```toml
+[all]
+enabled = true
+overridable = 2
+ignore = 'vendor1/'
+
+[section1]
+inherits = 'all'
+appends = 'ignore'
+overrideable = 3
+ignore = 'value2/'
+other = 'some_value'
+```
+
+in TOML
+
+
+### Trying to have syntax like coafile
```

Comment:
No need, we will follow the TOML standard format in most cases because current users of TOML will get confused, we will customize only when it is necessary.

(Leave this section here, don't remove)